app/models/course/discussion/post.rb

```diff
@@ -1,5 +1,17 @@
 class Course::Discussion::Post < ActiveRecord::Base
   acts_as_forest order: :created_at
 
+  after_initialize :set_topic, if: :new_record?
+  validate :topic_must_be_the_same_as_parent_post
+
   belongs_to :topic, inverse_of: :posts
+
+  def set_topic
+    self.topic = parent.topic if parent
+  end
+
+  def topic_must_be_the_same_as_parent_post
```

Comment:
Sorry, just found that you can do this in one line, and I think it's more clear:
```
errors.add(t('course.discussion.posts.validate_topic')) if parent && topic != parent.topic
```